src/core/model.jl

```diff
@@ -84,6 +84,33 @@ function AgentBasedModel(
     return ABM{A, S, F, P}(agents, space, scheduler, properties)
 end
 
+"""
+    AgentBasedModel(agent [, space]; scheduler, properties)
+Similar to the constructor above, but created via an instance of your agent.
+
+```julia
+mutable struct YourAgent <: AbstractAgent
+    id::Int
+    pos::Tuple{Int, Int}
+end
+agent = YourAgent(1, (1,1))
+model = AgentBasedModel(agent, GridSpace((10,10))
+```
+
+Note that the specific values you use when creating `agent` are not important and wont be used here.
+This process just helps Agents.jl do some performance checks for you, if you're not that familiar with
+Julia's type system.
+"""
+function AgentBasedModel(
+        agent::A, space::S = nothing;
+        scheduler::F = fastest, properties::P = nothing
+        ) where {A<:AbstractAgent, S<:SpaceType, F, P}
+    agent_validator(typeof(agent), space)
+
+    agents = Dict{Int, typeof(agent)}()
```

Comment:
```suggestion
    agents = Dict{Int, A}()
```

Here `A` and `typeof(agent)` are textually equivalent.

Reply:
Good to know. Thought that would be the case.